lib/queue.js

```diff
@@ -199,28 +217,36 @@ class Queue extends Emitter {
         clients.push(this.eclient);
       }
 
-      return Promise.all(clients.map((client) => {
-        const promise = helpers.deferred();
-        client.quit(promise.defer());
-        return promise;
-      }));
+      return Promise.all(
+        clients.map((client) => {
+          const promise = helpers.deferred();
+          client.quit(promise.defer());
+          return promise;
+        })
+      );
     };
 
-    const closed = helpers.withTimeout(this._ready.then(() => {
-      // Stop the blocking connection, ensures that we don't accept additional
-      // jobs while waiting for the ongoing jobs to terminate.
-      if (this.settings.isWorker) {
-        redis.disconnect(this.bclient);
-      }
-
-      // Wait for all the jobs to complete. Ignore job errors during shutdown.
-      const waitJobs = Array.from(this.activeJobs);
-      return Promise.all(waitJobs.map((promise) => promise.catch(() => {})));
-    }), timeout).then(() => {
-      return cleanup().then(() => undefined);
-    }, (err) => {
-      return cleanup().then(() => Promise.reject(err));
-    });
+    const closed = finally_(
+      helpers
+        .withTimeout(
+          this._ready.then(() => {
+            // Stop the blocking connection, ensures that we don't accept
+            // additional jobs while waiting for the ongoing jobs to terminate.
+            if (this.settings.isWorker) {
+              redis.disconnect(this.bclient);
+            }
+
+            // Wait for all the jobs to complete. Ignore job errors during
+            // shutdown.
+            return Promise.all(
+              Array.from(this.activeJobs, (promise) => promise.catch(() => {}))
+            );
+          }),
+          timeout
+        )
+        .then(() => {}),
+      () => cleanup()
```

Comment:
Potential unhandled rejection?

Reply:
`finally` actually waits for any `Promise` returned from the provided function!

Reply:
Resolved offline - solution was bbb56903828e4d22875dfa90d4f720ee27c47c29, and we don't anticipate the swallowing of subsequent errors to be problematic.